### ISSUE TITLE
`derive_partial_eq_without_eq` triggers inside binary crates

### ISSUE THREAD

**Issue by jplatte** [rust-lang/rust-clippy#9529]
### Summary

I don't think it should.

### Lint Name

derive_partial_eq_without_eq

### Reproducer

I tried this code:

```rust
#[derive(PartialEq)]
pub enum Explore {
    Stable,
    RecentlyStabilized,
    Unstable,
}
```

in a binary crate: I saw the `derive_partial_eq_without_eq` being triggered. I expected to not have it trigger.

### Version

```text
rustc 1.66.0-nightly (4a1467723 2022-09-23)
binary: rustc
commit-hash: 4a14677239bbd9a3892d165ee24513c1a05820e4
commit-date: 2022-09-23
host: x86_64-unknown-linux-gnu
release: 1.66.0-nightly
LLVM version: 15.0.0
```

**Comment by xFrednet** [comment]
I would actually expect it to trigger in binary crates as well. Adding a config value might be a good solution.

**Comment by jplatte** [comment]
Why is that?

**Comment by xFrednet** [comment]
Mainly because I see this as a best practice. Even if the bound is not required, adding it is simple. If it's later required, you don't have to change a different file.

**Comment by lovasoa** [comment]
I agree with the original poster. There are many traits that could potentially be implemented on my data structures, but if I don't need them, I am not going to add a derive for them. It would just add junk to the codebase, and make the compile times longer for no benefit.

**Comment by xFrednet** [comment]
Maybe a config value might be nice, but I wouldn't expect many people to enable is, if it's false by default. I'd be okay with this being changed. I don't have a strong opinion on this one :)
